gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/process/traversal/dsl/graph/GraphTraversal.java

```diff
@@ -1034,11 +1034,10 @@
      * @param toStepLabel the step label of the incoming vertex
      * @return the traversal with the modified {@link AddEdgeStep}
      * @since 3.1.0-incubating
-     * @see <a href="http://tinkerpop.apache.org/docs/${project.version}/reference/#addedge-step" target="_blank">Reference Documentation - AddEdge Step</a>
      */
     public default GraphTraversal<S, E> to(final String toStepLabel) {
         this.asAdmin().getBytecode().addStep(Symbols.to, toStepLabel);
-        ((AddEdgeStep) this.asAdmin().getEndStep()).addTo(__.select(toStepLabel));
+        ((FromToModulating) this.asAdmin().getEndStep()).addTo(toStepLabel);
```

Comment:
Now that `from()` is more generic the javadoc needs to be updated to better reflect usage. Also, looks like you removed the `@see` annotation. I think that should be added back with additional ones linking to the appropriate steps that make use of this. I didn't look at the changes for `to()` in here but I assume this comment would apply to those as well.